src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,83 @@
+package core.basesyntax.service;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.RegistrationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+    private static StorageDaoImpl storageDao;
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @Test
+    void register_user_isOK() {
+        User expected = new User("BobGood", "1234567", 18);
+        User actual = registrationService.register(expected);
+        assertEquals(actual, expected);
+    }
+
+    @Test
+    void register_userPasswordIsShort_isNotOK() {
+        User expected = new User("BobWithShortPassword", "12", 18);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(expected);
+        });
+    }
+
+    @Test
+    void register_userLoginIsShort_isNotOK() {
+        User expected = new User("Bob", "1234567", 18);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(expected);
+        });
+    }
+
+    @Test
+    void register_userUnderage_isNotOK() {
+        User expected = new User("BobChild", "1234567", 15);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(expected);
+        });
+    }
+
+    @Test
+    void register_userLoginIsNull_isNotOK() {
+        User expected = new User(null, "1234567", 18);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(expected);
+        });
+    }
+
+    @Test
+    void register_userPasswordIsNull_isNotOk() {
+        User expected = new User("BobWithoutPassword", null, 18);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(expected);
+        });
+    }
+
+    @Test
+    void register_userAgeIsNull_isNotOK() {
+        User expected = new User("BobWithoutAge", "1234567", null);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(expected);
+        });
+    }
+
+    @Test
+    void register_userAlreadyExists_isNotOK() {
+        User user = new User("BobAlredyExists", "john123", 28);
+        storageDao.add(user);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
```

Comment:
In this test, you are adding a user to the storage and then trying to register the same user again. This is not a good practice because your tests should be independent and not rely on each other to fill the data. Instead of adding the user to the storage directly, you should register a new user with the same login. This way, the test will not depend on the state of the storage before it is executed.